### ISSUE TITLE
Google cloud sql using wrong ssl cert to to establish socket connection from kubernetes

### ISSUE THREAD

**Issue by Jakedavies** [GoogleCloudPlatform/cloudsql-proxy#34]
We have 2 clusters, a staging and production cluster. We setup our production cluster to use cloud-sql as a pod attached to our app pods to connect to the cloud sql servers. 

When we stood up the staging cluster, everything worked except that our apps cloud-sql proxies are throwing this error, all the secrets and associated paths have been changed for the staging cluster.

`couldn't connect to " project-name:us-central1:name-db-staging": x509: certificate is valid for  project-name:name-db, not project-name:name-db-staging`

This is happening for two separate projects, both are saying the production ssl cert is being used, but the db it is trying to connect to is the staging server

**Comment by Carrotman42** [comment]
Interesting, I'm wondering whether something is messed up on the backend. Can you email your project name and instances to cloud-sql@google.com and I can help debug the issue from there?

Some debugging we can do in public, though: how are your staging/production databases set up? Is one created from a backup of another or something?

**Comment by Jakedavies** [comment]
The staging/production databases are essentially the exact same, and there is a nightly cron that takes the backup from the production and restores it to the staging database.

**Comment by Carrotman42** [comment]
To follow up, the workaround here was to reset the SSL configuration of the instance. I'm currently investigating the root cause of the issue.

**Comment by roberthbailey** [comment]
@Carrotman42 — Let me know if you need any help debugging from the GKE-side of things.

**Comment by Carrotman42** [comment]
Thanks for the offer, but it turned out to be a strictly Cloud SQL-related problem. I submitted a fix this morning and it should be in production within a week or so. There may still be bugs related to this one which are yet undiscovered, but until then the workaround of resetting SSL configuration should work for this class of bug.

I will close this bug when the fix finishes rolling out.

**Comment by Carrotman42** [comment]
The fix should be rolled out. Please reopen if you're still encountering the problem.

**Comment by retnuh** [comment]
Hi there, after creating a new instance and restoring from a backup of a different instance, we encountered the same issue.  Doing a reset of the SSL configuration cleared the problem.

**Comment by 4dex** [comment]
In case anyone else encounters this issue again, I came across it when using an older version of the cloud_sql_proxy image. Had to update the image before the proxy would connect to my cloud sql instances. 

Updated from `image: gcr.io/cloudsql-docker/gce-proxy:1.11` to `image: gcr.io/cloudsql-docker/gce-proxy:1.33.2` and the `x509: certificate is valid for  project-name:name-db, not project-name:name-db-whatever` issue went away.
